rest_controllers/AuthRestController.php

```diff
@@ -31,25 +31,32 @@ public function authenticate($authPayload)
             return;
         }
         if (!empty($_SESSION['api']) && !empty($_SESSION['site_id'])) {
-            $encoded_api = bin2hex(trim($_SESSION['api']));
-            $encoded_site = bin2hex(trim($_SESSION['site_id']));
+            $encoded_api_site = bin2hex(trim($_SESSION['api']) . trim($_SESSION['site_id']));
```

Comment:
will throw off byte count on decode if site_id is not bin2hex.

Reply:
To clarify, am still doing both bin2hex on both $_SESSION['api'] and $_SESSION['site_id']. But am just combining them first doing the bin2hex conversion.

Reply:
So then consistent with opposite of this:
https://github.com/openemr/openemr/blob/master/apis/dispatch.php#L50

Reply:
Oh, sorry Brady. Missed outer quotes. Looks okay to me then. Can you think of a simple way to verify a valid site-id without globals? I don't know that we put that in a table anywhere

Reply:
We can do that (by ensuring it is equal to one of the dir in sites/ . I'll look into this since planning to do a couple more things here (encrypt token stored in database and thinking about some mechanism to prevent brute force attacks on both user/pass and token vectors). Gonna keep working on this PR.